MovieViewer/MoviesSearchViewController.swift

```diff
@@ -0,0 +1,251 @@
+//
+//  MoviesSearchViewController.swift
+//  MovieViewer
+//
+//  Created by Dang Quoc Huy on 7/7/16.
+//  Copyright © 2016 Dang Quoc Huy. All rights reserved.
+//
+
+import UIKit
+import MBProgressHUD
+
+class MoviesSearchViewController: UIViewController {
+
+    @IBOutlet weak var tableView: UITableView!
+    @IBOutlet weak var errorView: UIView!
+    @IBOutlet weak var errorLabel: UILabel!
+    
+    var searchBar = UISearchBar()
+    var noDataLabel: UILabel!
+    var searchMovieSettings = SearchMovieSettings()
+    var movies: [Movie]? {
+        didSet {
+            
+            filteredMovies = movies
+        }
+    }
+    var filteredMovies: [Movie]?
+    var isMoreDataLoading = false
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        // set delegate
+        tableView.dataSource = self
+        tableView.delegate = self
+        searchBar.delegate = self
+        
+        // Add SearchBar to the NavigationBar
+        searchBar.sizeToFit()
+        navigationItem.titleView = searchBar
+        
+        // init UI theme
+        setTheme()
+        hideError()
+    }
+    
+    // In a storyboard-based application, you will often want to do a little preparation before navigation
+    override func prepareForSegue(segue: UIStoryboardSegue, sender: AnyObject?) {
+        
+        if segue.identifier == "filtersSegue" {
+            // we wrapped our FiltersViewController inside a UINavigationController
+            let navController = segue.destinationViewController as! UINavigationController
+            let filtersView = navController.topViewController as! FiltersViewController
+            filtersView.settings = self.searchMovieSettings
+        }
+        
+        if let cell = sender as? UITableViewCell {
+            let indexPath = tableView.indexPathForCell(cell)
+            let movie = filteredMovies![indexPath!.row]
+            let movieDetailView = segue.destinationViewController as! MovieDetailViewController
+            movieDetailView.movie = movie
+        }
+    }
+    
+    func numberOfSectionsInTableView(tableView: UITableView) -> Int {
+        
+        var numOfSection = 0
+        if let filteredMovies = filteredMovies {
+            if filteredMovies.count > 0 {
+                tableView.separatorStyle = .SingleLine
+                numOfSection = 1
+                tableView.backgroundView = nil
+            } else {
+                tableView.backgroundView = noDataLabel
+                tableView.separatorStyle = .None
+            }
+        }
+        
+        return numOfSection
+    }
+    
+    @IBAction func didSaveSettings(segue: UIStoryboardSegue) {
+        
+        let filtersView = segue.sourceViewController as! FiltersViewController
+        searchMovieSettings = filtersView.settings
```

Comment:
as my test, this line is...necessary.
Cause SearchMovieSettings is a struct, in order to get value back, I need to copy settings from FiltersViewController